src/com/library/backlog/Library.java

```diff
@@ -0,0 +1,194 @@
+package com.library.backlog;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import com.library.userroles.UserRole;
+
+public class Library {
+	
+	String isbn;
+	String title;
+	Book book;
+	ArrayList<User> userList;
+	private UserRole role;
+	String usernameEntered;
+	String passwordEntered;
+	private List<Book> bookList;
+	// actually we doing this here:
+	// List<Book> bookList = new ArrayList<Book>(); 
+	// List is an interface and ArrayList is the implementation of it, List gives more flexibility than an ArrayList and less bugs
+	
+	
+	public Library(){
+		
+		//the arraylist to which books can be added
+		bookList = new ArrayList<Book>();
+		userList = new ArrayList<User>();
+	}
+		
+	
+		//adding new book
+		public void addNewBook(){
+
+			@SuppressWarnings("resource")
+			Scanner reader = new Scanner(System.in);  // Reading from System.in
+
+			System.out.println("Enter book isbn:");
+			isbn = reader.nextLine();
+			System.out.println("Enter book title:");
+			title = reader.nextLine();
+			
+			Book newBook= new Book(isbn, title);
+			//book already exist or not
+			if(this.bookList.contains(newBook)){
+				System.out.println("The book already exists");
+			}
+			else{
+				bookList.add(newBook);
+			}
+		}//end of addNewBook()
+			
+		//searching for a book
+		public void search(){
+			
+			/**
+			 * TODO
+			 * prompt the user for a search when they want to search prompt them for title and isbn
+			 */
+			
+			@SuppressWarnings("resource")
+			Scanner reader = new Scanner(System.in);  // Reading from System.in
+			System.out.println("Enter book title:");
+			String searchTitle = reader.nextLine();
+			
+			System.out.println("Enter book isbn:");
+			String searchIsbn = reader.nextLine();
+			
+			Book searchBook = new Book(searchIsbn,searchTitle);
+			
+			if(bookList.contains(searchBook)){
+			
+			for(int i = 0; i <= bookList.size(); i++){
+				
+				if(searchIsbn.equalsIgnoreCase(book.getIsbn()) && searchTitle.equalsIgnoreCase(book.getTitle())){
+					
+					/**
+					 * TODO compare the object bookSearch with objects contained in the ArrayList
+					 * actually its not a TODO for I
+					 * 
+					 */
+					
+					if(!bookList.get(i).getcheckedOut()){
+						System.out.println("the book has been found and available");
+						}
+					
+					else{
+							System.out.println("The book has been assigned to someone else, please wait");
+							}
+					}
+				
+				else{
+					System.out.println("The book is not part of the library");
+					}
+				}
+			}
+
+		}//end of search ///////////////better would be if I use exception here book not found
+		
+		
+		
+		public void login(String inputUsername, String inputPassword){
+			
+			User userLogin = new User(User.getRole(), inputUsername, inputPassword);
```

Comment:
here you created a new object from the data that given from app user, you shouldn't do it like this 
replace line 104 with this
User userLogin=null;
for(User u : userList){
if(u.getUsername().equals(inputUsername) && u.getPassword().equals(inputPassword){
/*user exists and the object is..>>*/ userLogin=u;
}
}
replace line 106 >> if(userLogin != null && userLogin.getRole().equals(UserRole.ADMIN)){...} //directly
// then add 
else if(userLogin != null && userLogin.getRole().equals(UserRole.USER){...}//  the role is USER the else the user 
else{..} the user doesn't exist